File: repos_estudiantes_ALVAREZ_primera_entrega/APAZA_SEGA_LEONEL_ISAIAS/Introduccion/3.py
class Producto:
    def __init__(self, nombre, precio, stock):
        self.nombre = nombre
        self.precio = precio
        self.stock = stock

    def vender(self, cantidad):
        if cantidad <= self.stock:
            self.stock -= cantidad
            print(f"✅ Venta realizada de {cantidad} {self.nombre}(s).")
        else:
            print("❌ No hay suficiente stock.")

    def reabastecer(self, cantidad):
        self.stock += cantidad
        print(f"Se reabastecieron {cantidad} unidades.")

    def mostrar(self):
        print(f"{self.nombre} - Precio: {self.precio} Bs - Stock: {self.stock}")

# Ejemplo
p1 = Producto("Lapicero", 2.5, 10)
p1.vender(3)
p1.reabastecer(5)
p1.mostrar()

File: repos_estudiantes_ALVAREZ_primera_entrega/APAZA_SEGA_LEONEL_ISAIAS/Introduccion/3.java
public class Producto {
    private String nombre;
    private double precio;
    private int stock;

    public Producto(String nombre, double precio, int stock) {
        this.nombre = nombre;
        this.precio = precio;
        this.stock = stock;
    }

    public void vender(int cantidad) {
        if (cantidad <= stock) {
            stock -= cantidad;
            System.out.println("Venta realizada: " + cantidad + " unidades de " + nombre);
        } else {
            System.out.println("No hay suficiente stock disponible.");
        }
    }

    public void reabastecer(int cantidad) {
        stock += cantidad;
        System.out.println("Stock actualizado: " + stock);
    }

    public void mostrar() {
        System.out.println("Producto: " + nombre + " | Precio: " + precio + " | Stock: " + stock);
    }

    public static void main(String[] args) {
        Producto p1 = new Producto("Laptop", 5500.0, 10);
        p1.mostrar();
        p1.vender(3);
        p1.reabastecer(5);
        p1.mostrar();
    }
}

